Shopify/cli
# Adopt the CLI UI kit in the `shopify theme list` command

### WHY are these changes introduced?

Fixes:
* https://github.com/Shopify/internal-cli-foundations/issues/498

### WHAT is this pull request doing?

- Displays a table with all themes, following the CLI UI Kit convention
- Calls Admin API’s `/themes` endpoint with query parameter `fields`, to only retrieve fields `id`, `name`, `role`, for both `theme open` and `theme list` command
- Introduces `--role`, `--id` and `--name` parameters for `theme list` command, to filter for themes with the specific role/name/ID

### How to test your changes?

- Run `pnpm run shopify theme list`…
  - in dark mode: Verify that text is white/grey on dark background
  - in bright mode: Verify that text is black/grey on bright background
- Run `pnpm run shopify theme list --role=live` and verify that only `live` themes are listed
  - Run `pnpm run shopify theme list --role=ottawa` and verify that error `Expected --role=ottawa to be one of: live, unpublished, development` is displayed
- Given a theme named "My theme 12345", run `pnpm run shopify theme list --name="eMe 123"` and verify that your theme is listed
  - Run `pnpm run shopify theme list --name=(random string)` and verify that error `The … store doesn't have a theme with the "…" ID or name` is displayed
- Run `pnpm run shopify theme list --id=(ID of existing theme)` and verify that your theme is listed
  - Run command with non-existing ID and verify that error `The … store doesn't have a theme with the "…" ID or name` is displayed

Note: Multi-filter is not supported.
E.g.: `role` would rule over `id`, and `id` would rule over `name`.
```
shopify theme list --id=123 --name=a --role=unpublished

name    role           id
Studio  [unpublished]  #456
```

### Post-release steps

Update `shopify theme list` [docs](https://shopify.dev/themes/tools/cli/commands#list) with the new flags.


### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
